src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder.java

```diff
@@ -50,22 +54,91 @@ public class ChunkBuilder<T extends ChunkGraphicsState> {
+    // This is the initial number of threads we spin up upon ChunkBuilder creation.
+    // Default: 2-4 depending on render distance. *Always* less than hardLimitThreads.
+    private final int initialThreads;
+    // This is the number of threads we want to 'quickly' get up to. This is calculated
+    // by getOptimalThreadCount, but can also be configured by the user. Always less than hardLimitThreads.
+    private final int targetThreads;
+    // This is the number of threads we are allowed to create in total.
+    // This is defaulted to targetThreads, but maxes out at 64. Testing would be required to determine what
+    // actual count is optimal for a specific user and use case.
+    private final int hardLimitThreads;
+    // This is the initial time when this builder is created. We use this to create more threads.
+    private long initialTime;
+    // This is the user-configurable time delta to create a new thread, up until targetThreads.
+    private final long quickThreadCreationInterval;
+    // This is the user-configurable time delta to create a new thread, up until hardLimitThreads.
+    private final long slowThreadCreationInterval;
+
     private final ChunkVertexType vertexType;
     private final ChunkRenderBackend<T> backend;
 
-    public ChunkBuilder(ChunkVertexType vertexType, ChunkRenderBackend<T> backend) {
+    public ChunkBuilder(ChunkVertexType vertexType, ChunkRenderBackend<T> backend, int viewDistance) {
         this.vertexType = vertexType;
         this.backend = backend;
-        this.limitThreads = getOptimalThreadCount();
+
+        // User-configurable options for chunk threads.
+        int desiredTargetThreads = SodiumClientMod.options().advanced.targetChunkThreads;
+        int desiredInitialThreads = SodiumClientMod.options().advanced.initialChunkThreads;
+
+        // Yikes.
+        this.initialTime = currentTimeMillis();
+        this.quickThreadCreationInterval = MathHelper.clamp(SodiumClientMod.options().advanced.quickThreadCreationInterval, 1, 2000);
+        this.slowThreadCreationInterval = MathHelper.clamp(SodiumClientMod.options().advanced.slowThreadCreationInterval, 1, 60000);
+
+        // Our hard limit of threads. Cap at 64, prefer desiredMaxThreads, otherwise use optimal thread count.
+        this.hardLimitThreads = getMaxThreadCount();
+        // Our targeted number of threads.
+        this.targetThreads = MathHelper.clamp(desiredTargetThreads == 0 ? getOptimalThreadCount() : desiredTargetThreads, 1, this.hardLimitThreads);
+        // Our initial threads. A bit of a silly calculation for this one.
+        this.initialThreads = MathHelper.clamp(desiredInitialThreads == 0 ? (viewDistance / 10) + 2 : desiredInitialThreads, 1, this.targetThreads);
+    }
+
+    /**
+     * Returns the "optimal" number of threads to be used for chunk build tasks. This will always return at least one
+     * thread.
+     */
+    private static int getOptimalThreadCount() {
```

Comment:
imo it's a little confusing to reuse the "getOptimalThreadCount" method name for a different purpose here, maybe it would be better to just call it "getSuggestedTargetThreadCount" or similar to keep the naming system consistent between methods and fields?

Reply:
That's fair, I'll rename it to `getDefaultTargetThreads` to align